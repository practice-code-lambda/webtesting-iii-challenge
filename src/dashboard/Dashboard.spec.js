import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import Dashboard from './Dashboard'

describe('<Dashboard/>', () => {
    it('should take a snapshot', () => {
        const snap = renderer.create(<Dashboard />).toJSON();
        //Jest
        expect(snap).toMatchSnapshot()
    })
    it('should render open text', () => {
        const Dash = render(<Dashboard />)
        //Testing Library
        Dash.getByText(/open/i)
    })
    it('should show a closed gate', () => {

        const { getByText } = render(<Dashboard />)

        const button = getByText(/close gate/i)

        fireEvent.click(button)

        getByText(/closed/i)
    })
    it('should show a locked gate', () => {

        const { getByText } = render(<Dashboard />)

        const Closebutton = getByText(/close gate/i)
        const Lockbutton = getByText(/lock gate/i)

        fireEvent.click(Closebutton)
        fireEvent.click(Lockbutton)

        getByText(/locked/i)

    })
})


