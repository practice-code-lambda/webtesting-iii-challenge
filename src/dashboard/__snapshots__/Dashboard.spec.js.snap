// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> should take a snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="togglelock"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="toggleClosed"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
